sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -411,4 +327,372 @@ public void run() {
+       *       exported.
+       * </ul>
+       *
+       * @param configMap {@link Map} holding the configuration values.
+       * @return this.
+       */
+      @VisibleForTesting
+      Builder fromConfigMap(Map<String, String> configMap, NamingConvention namingConvention) {
+        configMap = namingConvention.normalize(configMap);
+        Long longValue = getLongProperty(KEY_SCHEDULE_DELAY_MILLIS, configMap);
+        if (longValue != null) {
+          this.setScheduleDelayMillis(longValue);
+        }
+        Integer intValue = getIntProperty(KEY_MAX_QUEUE_SIZE, configMap);
+        if (intValue != null) {
+          this.setMaxQueueSize(intValue);
+        }
+        intValue = getIntProperty(KEY_MAX_EXPORT_BATCH_SIZE, configMap);
+        if (intValue != null) {
+          this.setMaxExportBatchSize(intValue);
+        }
+        intValue = getIntProperty(KEY_EXPORT_TIMEOUT_MILLIS, configMap);
+        if (intValue != null) {
+          this.setExporterTimeoutMillis(intValue);
+        }
+        Boolean boolValue = getBooleanProperty(KEY_SAMPLED, configMap);
+        if (boolValue != null) {
+          this.setExportOnlySampled(boolValue);
+        }
+        return this;
+      }
+
+      /**
+       * Sets the configuration values from the given properties object for only the available keys.
+       * This method looks for the following keys:
+       *
+       * <ul>
+       *   <li>{@code otel.bsp.schedule.delay}: to set the delay interval between two consecutive
+       *       exports.
+       *   <li>{@code otel.bsp.max.queue}: to set the maximum queue size.
+       *   <li>{@code otel.bsp.max.export.batch}: to set the maximum batch size.
+       *   <li>{@code otel.bsp.export.timeout}: to set the maximum allowed time to export data.
+       *   <li>{@code otel.bsp.export.sampled}: to set whether only sampled spans should be
+       *       exported.
+       * </ul>
+       *
+       * @param properties {@link Properties} holding the configuration values.
+       * @return this.
+       */
+      public Builder readProperties(Properties properties) {
+        Long longValue = getLongProperty(KEY_SCHEDULE_DELAY_MILLIS, properties);
+        if (longValue != null) {
+          this.setScheduleDelayMillis(longValue);
+        }
+        Integer intValue = getIntProperty(KEY_MAX_QUEUE_SIZE, properties);
+        if (intValue != null) {
+          this.setMaxQueueSize(intValue);
+        }
+        intValue = getIntProperty(KEY_MAX_EXPORT_BATCH_SIZE, properties);
+        if (intValue != null) {
+          this.setMaxExportBatchSize(intValue);
+        }
+        intValue = getIntProperty(KEY_EXPORT_TIMEOUT_MILLIS, properties);
+        if (intValue != null) {
+          this.setExporterTimeoutMillis(intValue);
+        }
+        Boolean boolValue = getBooleanProperty(KEY_SAMPLED, properties);
+        if (boolValue != null) {
+          this.setExportOnlySampled(boolValue);
+        }
+        return this;
+      }
```

Comment:
Why do you not call `fromConfigMap`? Seems a lot of duplicated code. Do I miss anything? If yes maybe a comment would help others to also understand.

Reply:
`return fromConfigMap(Maps.fromProperties(properties), NamingConvention.DOT);`

Reply:
I overlooked the possibility to coerce a Property object into a Map!